projects/igniteui-angular/src/lib/grids/grid.rowEdit.directive.ts

```diff
@@ -1,29 +1,49 @@
 import { Directive, ElementRef, HostListener, Inject } from '@angular/core';
-import { GridType, IGX_GRID_BASE } from './common/grid.interface';
+import { GridType, IgxGridRowEditActionsTemplateContext, IgxGridRowEditTemplateContext, IgxGridRowEditTextTemplateContext, IGX_GRID_BASE } from './common/grid.interface';
 
 /** @hidden @internal */
 @Directive({
     selector: '[igxRowEdit]'
 })
-export class IgxRowEditTemplateDirective { }
+export class IgxRowEditTemplateDirective {
+    public static ngTemplateContextGuard(_directive: IgxRowEditTemplateDirective,
+        context: unknown): context is IgxGridRowEditTemplateContext { 
+        return true;
+    };
+ }
 
 /** @hidden @internal */
 @Directive({
     selector: '[igxRowEditText]'
 })
-export class IgxRowEditTextDirective { }
+export class IgxRowEditTextDirective {
+    public static ngTemplateContextGuard(_directive: IgxRowEditTextDirective,
+        context: unknown): context is IgxGridRowEditTextTemplateContext { 
+        return true;
+    };
+ }
 
 /** @hidden @internal */
 @Directive({
     selector: '[igxRowAddText]'
 })
-export class IgxRowAddTextDirective { }
+export class IgxRowAddTextDirective {
+    public static ngTemplateContextGuard(_directive: IgxRowAddTextDirective,
+        context: unknown): context is null { 
```

Comment:
I think that this should not be null but for example - 
![Screenshot 2022-09-19 at 16 16 49](https://user-images.githubusercontent.com/37667452/191026143-a1669785-d5fe-48f2-a09e-5a25b89cbea7.png)
Because:
1. We do set `$implicit = null` if we are adding row
2. We should be consistence for example with `IgxDragIndicatorIconDirective` where we do not set `implicit` and we do not set context at all but we have set that the type of the context is `IgxGridEmptyTemplateContext`